tools/demo-gui/demo-gui/sof_controller_engine.py

```diff
@@ -0,0 +1,197 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+import subprocess
+import os
+import signal
+import re
+import math
+
+# Global variables to store the aplay/arecord process, paused state, current file, detected device, volume control, and EQ numid
+aplay_process = None
+arecord_process = None
+paused = None
+current_file = None
+device_string = None
+volume_control = None
+eq_numid = None
+
+extra_audio_paths = []
+
+def initialize_device():
+    global device_string, volume_control, eq_numid
+
+    try:
+        output = subprocess.check_output(["aplay", "-l"], text=True, stderr=subprocess.DEVNULL)
+
+        match = re.search(r"card (\d+):.*\[.*sof.*\]", output, re.IGNORECASE)
+        if match:
+            card_number = match.group(1)
+            device_string = f"hw:{card_number}"
+            print(f"Detected SOF card: {device_string}")
+        else:
+            print("No SOF card found.")
+            raise RuntimeError("SOF card not found. Ensure the device is connected and recognized by the system.")
+
+        controls_output = subprocess.check_output(["amixer", f"-D{device_string}", "controls"], text=True, stderr=subprocess.DEVNULL)
+
+        volume_match = re.search(r"numid=(\d+),iface=MIXER,name='(.*Master GUI Playback Volume.*)'", controls_output)
+        if volume_match:
```

Comment:
@alexb3103 How about using `Master Playback Volume` here and avoid adding a new topology just for gui? This won't scale when we want to support other topologies.

Reply:
@alexb3103 I'm good to merge once we fix this string.

Reply:
Fixed, also fixed the name in the GUI standard pipeline.